src/go/configgenerator/routegen/backend.go

```diff
@@ -0,0 +1,98 @@
+package routegen
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/esp-v2/src/go/configgenerator/routegen/helpers"
+	"github.com/GoogleCloudPlatform/esp-v2/src/go/options"
+	"github.com/GoogleCloudPlatform/esp-v2/src/go/util"
+	"github.com/GoogleCloudPlatform/esp-v2/src/go/util/httppattern"
+	routepb "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	servicepb "google.golang.org/genproto/googleapis/api/serviceconfig"
+)
+
+// BackendGenerator is a RouteGenerator to configure routes to the local
+// or remote backend service.
+type BackendGenerator struct {
+	HTTPPatterns             *httppattern.MethodSlice
+	BackendClusterBySelector map[string]*BackendClusterSpecifier
+	BackendRouteGen          *helpers.BackendRouteGenerator
+}
+
+// NewBackendRouteGensFromOPConfig creates BackendGenerator
+// from OP service config + descriptor + ESPv2 options.
```

Comment:
nit: is descriptor used for route generator?

Reply:
No. I guess service config contains descriptor internally, but i'll remove it from the comment.